pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/channel/MailboxContentObserver.java

```diff
@@ -62,22 +58,9 @@ public void onNext(MailboxContent mailboxContent) {
       _mailbox = _mailboxService.getReceivingMailbox(mailboxId);
     }
     try {
-      TransferableBlock block;
-      DataBlock dataBlock = DataBlockUtils.getDataBlock(mailboxContent.getPayload().asReadOnlyByteBuffer());
-      if (dataBlock instanceof MetadataBlock) {
-        Map<Integer, String> exceptions = dataBlock.getExceptions();
-        if (exceptions.isEmpty()) {
-          block = TransferableBlockUtils.getEndOfStreamTransferableBlock(((MetadataBlock) dataBlock).getStats());
-        } else {
-          _mailbox.setErrorBlock(TransferableBlockUtils.getErrorTransferableBlock(exceptions));
-          return;
-        }
-      } else {
-        block = new TransferableBlock(dataBlock);
-      }
-
```

Comment:
This code has been moved to `ReceivingMailbox.offerRaw` in order to be able to capture the stats there. Also, I think the responsibility of doing this should be in ReceivingMailbox and not in the pipeline itself.